### ISSUE TITLE
remove deprecated listen callback without accept or reject

### ISSUE THREAD

**Issue by timaschew** [deepstreamIO/deepstream.io-client-js#212]
In version 1.1 both functions has been deprecated:

`client.event.listen(pattern, function(eventName, isSubscribed) {})`
and
`client.record.listen(pattern, function(match, isSubscribed){})`

To do it the proper way you need to pass a third argument to the callback which provides two functions: `accept` and `reject`. One of them needs to be called within the callback.

``` javascript
listen(pattern, function(name, isSubscribed, response) {
  if (isSubscribed) {
    response.accept() // if you want to provide the event name / record name
    // otherwise call response.reject()
  } else {
    // stop providing the event name / record name
  }
})
```

For records the subscriber can check a boolean flag via `aRecord.hasProvider` or register a handler for the `hasProviderChanged` event for the same boolean flag.

For more details [checkout the documentation](http://deepstream.io/docs/client-js/datasync-client-record/#client-record-listen-pattern-callback-)

**Comment by ronag** [comment]
Does this really need to be deprecated though? If `arguments.length === 2` auto accept?

**Comment by timaschew** [comment]
This could change the behaviour of your app since the feature was a bit broken before.
To avoid breaking changes we decided to keep the old behaviour and add a deprecated message instead.

**Comment by AlexBHarley** [comment]
This has been resolved in the 2.0 release
